File: Backend.NET/Models/BookingHeaderTable.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FleetManagementSystem.Api.Models;

[Table("booking_header_table")]
public class BookingHeaderTable
{
    [Key]
    [Column("booking_id")]
    public long BookingId { get; set; }

    [Column("booking_date")]
    public DateTime? BookingDate { get; set; }

    [Column("confirmation_number")]
    public string? ConfirmationNumber { get; set; }

    [Column("booking_status")]
    public string? BookingStatus { get; set; }

    [ForeignKey("Customer")]
    [Column("cust_id")]
    public int CustomerId { get; set; }
    public CustomerMaster? Customer { get; set; }

    [Column("start_date")]
    public DateTime? StartDate { get; set; }

    [Column("end_date")]
    public DateTime? EndDate { get; set; }

    [ForeignKey("PickupHub")]
    [Column("pickup_location_id")]
    public int? PickupHubId { get; set; }
    public HubMaster? PickupHub { get; set; }

    [ForeignKey("ReturnHub")]
    [Column("return_hub_id")]
    public int? ReturnHubId { get; set; }
    public HubMaster? ReturnHub { get; set; }

    [ForeignKey("CarType")]
    [Column("cartype_id")]
    public long? CarTypeId { get; set; }
    public CarTypeMaster? CarType { get; set; }

    [ForeignKey("Car")]
    [Column("car_id")]
    public int? CarId { get; set; }
    public CarMaster? Car { get; set; }

    [Column("first_name")]
    public string? FirstName { get; set; }

    [Column("last_name")]
    public string? LastName { get; set; }

    [Column("address")]
    public string? Address { get; set; }

    [Column("state")]
    public string? State { get; set; }

    [Column("pin")]
    public string? Pin { get; set; }

    [Column("daily_rate")]
    public double? DailyRate { get; set; }

    [Column("weekly_rate")]
    public double? WeeklyRate { get; set; }

    [Column("monthly_rate")]
    public double? MonthlyRate { get; set; }

    [Column("email_id")]
    public string? EmailId { get; set; }

    [Column("book_car")]
    public string? Bookcar { get; set; }

    [Column("pickup_time")]
    public DateTime? PickupTime { get; set; }

    [Column("pickup_fuel_status")]
    public string? PickupFuelStatus { get; set; }

    [Column("pickup_condition")]
    public string? PickupCondition { get; set; }

    [Column("return_time")]
    public DateTime? ReturnTime { get; set; }

    [Column("return_fuel_status")]
    public string? ReturnFuelStatus { get; set; }

    [Column("return_condition")]
    public string? ReturnCondition { get; set; }
}

File: Backend.JAVA/src/main/java/com/example/demo/Entity/BookingHeaderTable.java
package com.example.demo.Entity;

import java.time.LocalDate;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "booking_header_table")
@Data
public class BookingHeaderTable {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "booking_id")
	private long bookingId;

	@Column(name = "booking_date")
	private LocalDate bookingDate;

	@Column(name = "confirmation_number", unique = true)
	private String confirmationNumber;

	@Column(name = "booking_status")
	private String bookingStatus; // CONFIRMED, CANCELLED, ALLOTTED, COMPLETED

	@ManyToOne
	@JoinColumn(name = "cust_id", referencedColumnName = "cust_id", nullable = false)
	private CustomerMaster customer;

	@Column(name = "start_date")
	private LocalDate startDate;

	@Column(name = "end_date")
	private LocalDate endDate;

	@ManyToOne
	@JoinColumn(name = "pickup_location_id") // Maps to hub_master
	private HubMaster pickupHub; // Mapped from logic, usually implicit in car but needed for search

	@ManyToOne
	@JoinColumn(name = "return_hub_id")
	private HubMaster returnHub;

	@ManyToOne
	@JoinColumn(name = "cartype_id")
	private CarTypeMaster carType;

	@ManyToOne
	@JoinColumn(name = "car_id")
	private CarMaster car;

	@Column(name = "first_name")
	private String firstName;

	@Column(name = "last_name")
	private String lastName;

	@Column(name = "address")
	private String address;

	@Column(name = "state")
	private String state;

	@Column(name = "pin")
	private String pin;

	@Column(name = "daily_rate")
	private Double dailyRate;

	@Column(name = "weekly_rate")
	private Double weeklyRate;

	@Column(name = "monthly_rate")
	private Double monthlyRate;

	@Column(name = "email_id")
	private String emailId;

	public long getBookingId() {
		return bookingId;
	}

	public void setBookingId(long bookingId) {
		this.bookingId = bookingId;
	}

	public LocalDate getBookingDate() {
		return bookingDate;
	}

	public void setBookingDate(LocalDate bookingDate) {
		this.bookingDate = bookingDate;
	}

	public CustomerMaster getCustomer() {
		return customer;
	}

	public void setCustomer(CustomerMaster customer) {
		this.customer = customer;
	}

	public LocalDate getStartDate() {
		return startDate;
	}

	public void setStartDate(LocalDate startDate) {
		this.startDate = startDate;
	}

	public LocalDate getEndDate() {
		return endDate;
	}

	public void setEndDate(LocalDate endDate) {
		this.endDate = endDate;
	}

	public CarTypeMaster getCarType() {
		return carType;
	}

	public void setCarType(CarTypeMaster carType) {
		this.carType = carType;
	}

	public CarMaster getCar() {
		return car;
	}

	public void setCar(CarMaster car) {
		this.car = car;
	}

	public String getFirstName() {
		return firstName;
	}

	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}

	public String getLastName() {
		return lastName;
	}

	public void setLastName(String lastName) {
		this.lastName = lastName;
	}

	public String getAddress() {
		return address;
	}

	public void setAddress(String address) {
		this.address = address;
	}

	public String getState() {
		return state;
	}

	public void setState(String state) {
		this.state = state;
	}

	public String getPin() {
		return pin;
	}

	public void setPin(String pin) {
		this.pin = pin;
	}

	public Double getDailyRate() {
		return dailyRate;
	}

	public void setDailyRate(Double dailyRate) {
		this.dailyRate = dailyRate;
	}

	public Double getWeeklyRate() {
		return weeklyRate;
	}

	public void setWeeklyRate(Double weeklyRate) {
		this.weeklyRate = weeklyRate;
	}

	public Double getMonthlyRate() {
		return monthlyRate;
	}

	public void setMonthlyRate(Double monthlyRate) {
		this.monthlyRate = monthlyRate;
	}

	public String getEmailId() {
		return emailId;
	}

	public void setEmailId(String emailId) {
		this.emailId = emailId;
	}

	public String getBookcar() {
		return Bookcar;
	}

	public void setBookcar(String bookcar) {
		Bookcar = bookcar;
	}

	@Column(name = "book_car")
	private String Bookcar;

	@Column(name = "pickup_time")
	private java.time.LocalDateTime pickupTime;

	@Column(name = "pickup_fuel_status")
	private String pickupFuelStatus;

	@Column(name = "pickup_condition")
	private String pickupCondition;

	@Column(name = "return_time")
	private java.time.LocalDateTime returnTime;

	@Column(name = "return_fuel_status")
	private String returnFuelStatus;

	@Column(name = "return_condition")
	private String returnCondition;

	// Manual Getters and Setters for missing fields

	public String getConfirmationNumber() {
		return confirmationNumber;
	}

	public void setConfirmationNumber(String confirmationNumber) {
		this.confirmationNumber = confirmationNumber;
	}

	public String getBookingStatus() {
		return bookingStatus;
	}

	public void setBookingStatus(String bookingStatus) {
		this.bookingStatus = bookingStatus;
	}

	public HubMaster getPickupHub() {
		return pickupHub;
	}

	public void setPickupHub(HubMaster pickupHub) {
		this.pickupHub = pickupHub;
	}

	public HubMaster getReturnHub() {
		return returnHub;
	}

	public void setReturnHub(HubMaster returnHub) {
		this.returnHub = returnHub;
	}

	public java.time.LocalDateTime getPickupTime() {
		return pickupTime;
	}

	public void setPickupTime(java.time.LocalDateTime pickupTime) {
		this.pickupTime = pickupTime;
	}

	public String getPickupFuelStatus() {
		return pickupFuelStatus;
	}

	public void setPickupFuelStatus(String pickupFuelStatus) {
		this.pickupFuelStatus = pickupFuelStatus;
	}

	public String getPickupCondition() {
		return pickupCondition;
	}

	public void setPickupCondition(String pickupCondition) {
		this.pickupCondition = pickupCondition;
	}

	public java.time.LocalDateTime getReturnTime() {
		return returnTime;
	}

	public void setReturnTime(java.time.LocalDateTime returnTime) {
		this.returnTime = returnTime;
	}

	public String getReturnFuelStatus() {
		return returnFuelStatus;
	}

	public void setReturnFuelStatus(String returnFuelStatus) {
		this.returnFuelStatus = returnFuelStatus;
	}

	public String getReturnCondition() {
		return returnCondition;
	}

	public void setReturnCondition(String returnCondition) {
		this.returnCondition = returnCondition;
	}

}
